assets/js/wpjm-stats.js

```diff
@@ -118,11 +108,134 @@ import { createHooks } from '@wordpress/hooks';
 		}, {} );
 
 		Object.keys( statsByTrigger ).forEach( function ( triggerName) {
-			hookStatsForTrigger( statsByTrigger, triggerName );
+			WPJMStats.hookStatsForTrigger( statsByTrigger, triggerName );
+		} );
+
+		WPJMStats.initCallbacks.forEach( function ( initCallback ) {
+			initCallback.call( null );
+		} );
+
+		WPJMStats.hooks.doAction( 'page-load' );
+	},
+
+	hookStatsForTrigger: function ( statsByTrigger, triggerName ) {
+		console.log( 'hookStatsForTrigger' );
+		const statsToRecord    = [];
+		const stats            = statsByTrigger[triggerName] || [];
+		const events           = {};
+		const statsByType      = {};
+
+		stats.forEach( function ( statToRecord ) {
+			if ( ! statsByType[statToRecord.type] ) {
+				statsByType[statToRecord.type] = [];
+			}
+
+			statsByType[statToRecord.type].push( statToRecord );
+			statsToRecord.push( statToRecord );
 		} );
 
-		// Kick things off.
-		console.log('kick thing off');
-		window.wpjmStatHooks.doAction( 'page-load' );
+		// Hook action to call logStats.
+		WPJMStats.hooks.addAction( triggerName, 'wpjm-stats', function () {
+			window.wpjmLogStats( statsToRecord );
+		}, 10 );
+
+		Object.keys( statsByType ).forEach( function ( type ) {
+			console.log( 'stats by type', type );
+			WPJMStats.types[type] && WPJMStats.types[type]( statsByType[type] );
+		} );
+	},
+
+	hooks: createHooks(),
+	types: {
+		pageLoad: function ( stats ) {
+			console.log( 'pageLoad init script.' );
+			// This does not need to do anything special.
+		},
+		domEvent: function ( stats ) {
+			console.log( 'pageLoad init script.' );
+			const events = {};
+			stats.forEach( function ( statToRecord ) {
+				const triggerName = statToRecord.trigger;
+				if ( statToRecord.element && statToRecord.event ) {
+					const elemToAttach = document.querySelector( statToRecord.element );
+					if ( elemToAttach && ! events[statToRecord.element] ) {
+						elemToAttach.addEventListener( statToRecord.event, function ( e ) {
+							if ( checkUniqueRecordedToday( statToRecord ) ) {
```

Comment:
This check is duplicated since it's also part of `wpjmLogStats` now. 

To clarify what  I suggested in the other PR, we don't really need to wait to the request finish to call `setUniques`. We could filter the stats to log in the addAction callback, and have some logic like this for uniques:
- is this the first record today? then let it through to log and **mark it as logged immediately**
- otherwise remove it from the array

It's probably better if the unique check operation is atomic, otherwise, if a user quickly clicks the apply button twice, before the first log request is finished, we'd count it as two unique events.